### ISSUE TITLE
potential-bugs UselessPostfixExpression does not work in all cases

### ISSUE THREAD

**Issue by kepocnhh** [detekt/detekt#4617]
```kotlin
fun foo() {
    var i = 0
    i = i-- // work
    i = 1 + i++ // work
    i = i++ + 1 // work
    val i2 = i++ // does not work
    if (i2 == 1) { error("One!") }
    fun baz(): Int {
        var j = 0
        return j++ // work
    }
}
```

## Expected Behavior
The rule will work and the detekt will report an error in case of `val i2 = i++`.

## Observed Behavior
Detekt does not report an error in case of `val i2 = i++`.

## Steps to Reproduce
config:
```yml
build:
  maxIssues: 0

processors:
  active: false

console-reports:
  active: false

potential-bugs:
  active: true
  UselessPostfixExpression:
    active: true
```

## Context
Detekt reacts to the cases specified in the examples in the documentation. But it does not respond to the case specified above. Although even the ide suggests in this place.
![photo_2022-03-07_19-33-18](https://user-images.githubusercontent.com/6218886/157076728-97eec632-5473-4da0-b7c0-6711877b5115.jpg)

## Your Environment
detekt 1.20.0-RC1
Gradle 7.3
Kotlin 1.5.31
JVM 12.0.2 (Oracle Corporation 12.0.2+10)
OS: Linux 4.19.0-16-amd64

**Comment by BraisGabin** [comment]
It works as expected. We even have a test for that case to ensure that we don't report it.

Your example could be simplified to this:

```kotlin
fun foo() {
  var i = 0
  i++ // This one is also flagged by intellij
}
```

But this is not an useless postfix expression. This is a change in the value of `i` that is never used. We could have a rule to track this kind of things. But it should be more generic for example it should detect things like this:

```kotlin
var foo = "Hello"
println(foo)
foo = "bye" // We set this value but we never use it
```

And maybe that rule could include what this rule does.

**Comment by marschwar** [comment]
I think I agree with @kepocnhh here. I see no reason why those two cases should be handled any differently.

```kotlin
var i = 1
i = 1 + i++
println(i)

val j = 1 + i++
println(j)
```

**Comment by BraisGabin** [comment]
Because they do different things. I mean, I agree that those two codes should be flagged. But no by this rule. We should have another rule for it. Something similar as unused private member but for asignations.

**Comment by cortinico** [comment]
> Something similar as unused private member but for asignations.

+1 on this. Dropping the `IgnoredReturnValue` here as I believe is really similar to this case
